### PR TITLE
Test for LocalScripts plugin

### PR DESCRIPTION
**Pull request by stefan-caraiman**

### CODE REVIEW

File: argus/tests/cloud/windows/test_smoke.py [hunk 1]
```diff
@@ -82,6 +82,13 @@ def test_https_winrm_configured(self):
         stdout = remote_client.run_command_verbose('echo 1')
         self.assertEqual('1', stdout.strip())
 
+    def test_wmi_execution(self):
+        """Test if the localscripts plugin has executed
+         the WMI commands successfully."""
+        cmd = ('powershell "Get-Content C:\\Scripts\\wmi_log.txt"')
```
Comment (alexcoman) [inline argus/tests/cloud/windows/test_smoke.py]: Try: `command =  'powershell "Get-Content C:\\Scripts\\wmi_log.txt"'`

File: argus/resources/windows/test_wmi.ps1 [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#ps1_sysnative
+$ErrorOccured = $false
+
+try 
+{
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1]: Open braces on the same line. So it should be `try {`

File: argus/resources/windows/test_wmi.ps1 [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+$ErrorOccured = $false
+
+try 
+{ 
+   $ErrorActionPreference = 'Stop'
+   (Get-WmiObject Win32_OperatingSystem).OperatingSystemSKU -notmatch 0
+   # Checks if its a valid Windows version
+}
+catch
+{
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1]: ditto

File: argus/resources/windows/test_wmi.ps1 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+   Set-Content 'c:\\Scripts\\wmi_log.txt' 'F'
+   $ErrorOccured=$true
+}
+
+if(!$ErrorOccured) 
+{
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1]: ditto

File: argus/resources/windows/test_wmi.ps1 [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#ps1_sysnative
+$ErrorOccured = $false
+
+try 
+{ 
+   $ErrorActionPreference = 'Stop'
+   (Get-WmiObject Win32_OperatingSystem).OperatingSystemSKU -notmatch 0
+   # Checks if its a valid Windows version
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1:6]: Comments should be above the line they are referring.

File: argus/resources/windows/test_wmi.ps1 [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+}
+catch
+{
+   Set-Content 'c:\\Scripts\\wmi_log.txt' 'F'
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1]: Use full parameter names. It should be: `Set-Content -Path C:\Scripts\wmi_log.txt -Value 'F'`

File: argus/resources/windows/test_wmi.ps1 [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+   $ErrorOccured=$true
+}
+
+if(!$ErrorOccured) 
+{
+    Set-Content 'c:\\Scripts\\wmi_log.txt' 'T'
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1]: ditto

**Comment by alexcoman** [comment]
You should take a look on the following resources:
- [recipes/cloud/windows.py](https://github.com/cloudbase/cloudbase-init-ci/blob/eb762f42ee36c0c620d437f1856bc89e4437c4c6/argus/recipes/cloud/windows.py#L294)
- [installCBinit.ps1](https://github.com/cloudbase/cloudbase-init-ci/blob/master/argus/resources/windows/installCBinit.ps1#L11)

File: argus/resources/windows/test_wmi.ps1 [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+}
```
Comment (costingalan) [inline argus/resources/windows/test_wmi.ps1:16]: You need to end with a blank line.

File: argus/tests/cloud/windows/test_smoke.py [hunk 9]
```diff
@@ -82,6 +82,13 @@ def test_https_winrm_configured(self):
         stdout = remote_client.run_command_verbose('echo 1')
         self.assertEqual('1', stdout.strip())
 
+    def test_wmi_execution(self):
+        """Test if the localscripts plugin has executed
+         the WMI commands successfully."""
+        cmd = 'powershell "Get-Content C:\\Scripts\\wmi_log.txt"'
```
Comment (costingalan) [inline argus/tests/cloud/windows/test_smoke.py]: Here Get-Content should have the `-Path` parameter.

File: argus/recipes/cloud/windows.py [hunk 10]
```diff
@@ -203,6 +203,19 @@ def replace_code(self):
         command = '"{}" -m pip install -r C:\\cloudbaseinit\\requirements.txt'
         self._execute(command.format(python))
 
+    def _add_localscripts_file(self):
+        """Add various scripts to the localscripts folder,
+        in order to later test if the cloudbase plugin is
```
Comment (costingalan) [inline argus/recipes/cloud/windows.py]: I think it should be: `in order to later test if the cloudbase-init localscripts plugin is`

**Pull request closed without merging**